addons/netfox/state-synchronizer.gd

```diff
@@ -30,16 +30,64 @@ func _ready():
 func _after_tick(_dt, tick):
 	if is_multiplayer_authority():
 		# Submit snapshot
-		var state = PropertySnapshot.extract(_props)
-		_submit_state.rpc(state, tick)
+		var local_state: Dictionary = PropertySnapshot.extract(_props)
+		var properties_to_ignore: Array[String] = determine_properties_to_ignore(local_state)
+	
+		update_last_state_cache(local_state, tick)
+
+		var state_to_broadcast: Dictionary
+		if (NetworkRollback.enable_state_diffs):
+			state_to_broadcast = local_state.duplicate(true)
```

Comment:
idk if that boolean should be true or not. As it is (true) right now, it will definitely work as intended, but if it's needless, it should have a minor performance cost.